libp2p/NodeTable.cpp

```diff
@@ -199,7 +199,16 @@ void NodeTable::doDiscover(NodeId _node, unsigned _round, shared_ptr<set<shared_
 	m_timers.schedule(c_reqTimeout.count() * 2, [this, _node, _round, _tried](boost::system::error_code const& _ec)
 	{
 		if (_ec)
-			clog(NodeTableWarn) << "Discovery timer canceled!";
+			clog(NodeTableWarn) << "Discovery timer canceled: " << _ec.value() << _ec.message();
+
+		if (995 == _ec.value() || m_timers.isStopped())
```

Comment:
indeed. right now this just looks like a backdoor!

also, while i'm aware of the reason for the awkward <constant> == <variable> ordering, the rest of the codebase uses <variable> == <constant> ordering. consistency is king.